src/validation.cpp

```diff
@@ -4051,9 +4051,13 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
     if (pindexFailure)
         return error("VerifyDB(): *** coin database inconsistencies found (last %i blocks, %i good transactions before that)\n", chainActive.Height() - pindexFailure->nHeight + 1, nGoodTransactions);
 
+    // sub-level 3: update pindexState
+    if (nCheckLevel < 3) {
+        pindexState = pindex;
+    }
     // check level 4: try reconnecting blocks
     if (nCheckLevel >= 4) {
-        CBlockIndex *pindex = pindexState;
+        pindex = pindexState;
```

Comment:
Since you moved `pindex` to the outer scope and made it visible here, they are already the same (both of them are set to pindex->pprev at the end of the previous for loop)

At this point pindexState is only used to calculate the number of blocks, so it could make sense to just remove it altogether and calculate the number of blocks through a index_lowest or num_blocks.